app/database/manager/__mocks__/index.ts

```diff
@@ -66,367 +42,309 @@ if (__DEV__) {
-      } else {
-      // Or, it might be that the user has this server on the web-app but not mobile-app; so we'll need to create a new entry for this new serverUrl
-          const databaseName = connectionName ?? urlParse(serverUrl).hostname;
-          connection = await this.createDatabaseConnection({
-              shouldAddToDefaultDatabase: true,
-              configs: {
-                  actionsEnabled: true,
-                  dbName: databaseName,
-                  dbType: DatabaseType.SERVER,
-                  serverUrl,
-              },
-          });
-      }
-
-      if (setAsActiveDatabase) {
-          await this.setActiveServerDatabase(serverUrl);
-      }
-
-      return connection;
-  };
-
-  /**
-   * createDatabaseConnection: Creates database connection and registers the new connection into the default database.  However,
-   * if a database connection could not be created, it will return undefined.
-   * @param {DatabaseConfigs} databaseConnection
-   * @param {boolean} shouldAddToDefaultDatabase
-   *
-   * @returns {Promise<DatabaseInstance>}
-   */
-  createDatabaseConnection = async ({configs, shouldAddToDefaultDatabase = true}: DatabaseConnectionArgs): Promise<DatabaseInstance> => {
-      const {actionsEnabled = true, dbName = DEFAULT_DATABASE, dbType = DatabaseType.DEFAULT, serverUrl = undefined} = configs;
+    public appDatabase?: AppDatabase;
+    public serverDatabases: ServerDatabases = {};
+    private readonly defaultModels: Models;
+    private readonly iOSAppGroupDatabase: string | null;
+    private readonly androidFilesDirectory: string | null;
+    private readonly serverModels: Models;
+
+    constructor() {
+        this.defaultModels = [Info, Global, Servers];
+        this.serverModels = [
+            Channel,
+            ChannelInfo,
+            ChannelMembership,
+            CustomEmoji,
+            Draft,
+            File,
+            Group,
+            GroupMembership,
+            GroupsInChannel,
+            GroupsInTeam,
+            MyChannel,
+            MyChannelSettings,
+            MyTeam,
+            Post,
+            PostMetadata,
+            PostsInChannel,
+            PostsInThread,
+            Preference,
+            Reaction,
+            Role,
+            SlashCommand,
+            System,
+            Team,
+            TeamChannelHistory,
+            TeamMembership,
+            TeamSearchHistory,
+            TermsOfService,
+            User,
+        ];
+        this.iOSAppGroupDatabase = null;
+        this.androidFilesDirectory = null;
+    }
+
+    public init = async (serverUrls: string[]): Promise<void> => {
+        await this.initAppDatabase();
+        for await (const serverUrl of serverUrls) {
+            await this.initServerDatabase(serverUrl);
+        }
+    };
+
+    private createDatabase = async ({config, shouldAddToAppDatabase = true}: CreateDatabaseArgs): Promise<Database|undefined> => {
```

Comment:
Given that the `createDatabase` is used to create App db only, this method can further be cleaned.